src/app/campaign-list/card/campaign-flight-targets.pipe.ts

```diff
@@ -7,10 +7,10 @@ import { Flight } from '../campaign-list.service';
 export class CampaignFlightTargetsPipe implements PipeTransform {
   transform(flights: Flight[]): string {
     if (flights && flights.some(flight => flight.targets && flight.targets.length > 0)) {
-      return flights.reduce((acc, flight) => {
-        return acc += flight.targets && (acc && ', ') +
-          flight.targets.map(t => t.label).join(', ');
-      }, '');
+      const targets = flights.reduce((fAcc, flight) => {
+        return {...fAcc, ...flight.targets.reduce((tAcc, target) => ({...tAcc, [target.label]: target.label}), {})};
```

Comment:
Oof ... double reducers!  I guess there's no good way around it ... just wish typescript had a better utility for doing this.

Reply:
yeah, a little ugly. we could turn it into a stream and use distinct on it, stream all the things! yeah, maybe this is fine, I'm getting carried away